### PR TITLE
ci: build and publish daed docker image with geosite/geoip

### PR DESCRIPTION
**Pull request by kunish**
##  Description

This PR adds a dedicated workflow file to support `build and publish daed docker image with geosite/geoip` within github ci

## Note

Only build linux/amd64 for now

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -106,27 +103,20 @@ jobs:
         with:
           repository: daeuniverse/dae-wing
           path: dae-wing
-          ref: 'd986b18e10a7bfb90da1c15094b97e9386c459ef'
```
Comment (mzz2017) [inline .github/workflows/build.yml:109]: Why remove version pinning?

File: .github/workflows/publish-docker-image.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+jobs:
+  publish-docker-image:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: docker image - quay.io
+        uses: aevea/action-kaniko@master
+        with:
```
Comment (mzz2017) [inline .github/workflows/publish-docker-image.yml:17]: Does it support multiple architectures?

File: Dockerfile [hunk 3]
```diff
@@ -9,11 +9,26 @@ RUN corepack prepare pnpm@latest --activate
...
+FROM golang:1.20-bullseye as build-bundle
 
-WORKDIR /srv
+WORKDIR /build
+
+COPY --from=build-web /build/dist web
+
+RUN apt-get update && apt-get install -y llvm clang git make
+RUN git clone https://github.com/daeuniverse/dae-wing.git
+RUN cd dae-wing && make OUTPUT=daed WEB_DIST=/build/web/ bundle
```
Comment (mzz2017) [inline Dockerfile:20]: To consider cross-image compatibility, CGO_ENALBED=0 is necessary in most cases, or just keep in the same image with builder and remove useless pkgs.

File: .github/workflows/publish-docker-image.yml (hunk 2 above)
Comment (kunish) [reply .github/workflows/publish-docker-image.yml:17]: Good point, not for now

File: .github/workflows/publish-docker-image.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+          registry: quay.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.QUAY_PASS }}
+          image: ${{ github.repository }}
+          build_file: Dockerfile
+          tag_with_latest: true
+          cache: true
+          cache_registry: quay.io/${{ github.repository_owner }}/cache
```
Comment (yqlbu) [inline .github/workflows/publish-docker-image.yml:25]: Just a quick note: with this setup, Kaniko can ONLY build one `arch-based` image at a time.

File: Dockerfile [hunk 5]
```diff
@@ -9,11 +9,26 @@ RUN corepack prepare pnpm@latest --activate
 RUN pnpm install
 RUN pnpm build
 
-FROM caddy:alpine
+FROM golang:1.20-bullseye as build-bundle
```
Comment (yqlbu) [inline Dockerfile:12]: We may consider using `debian-12` as base image

File: Dockerfile [hunk 6]
```diff
@@ -9,11 +9,26 @@ RUN corepack prepare pnpm@latest --activate
...
 
-WORKDIR /srv
+WORKDIR /build
+
+COPY --from=build-web /build/dist web
+
+RUN apt-get update && apt-get install -y llvm clang git make
```
Comment (yqlbu) [inline Dockerfile:18]: Better add `cleanup` step afterwards. Ref: https://github.com/daeuniverse/container-estate/blob/master/self-runner-daed-dockerfile#L13-L15

File: .github/workflows/build.yml (hunk 1 above)
Comment (kunish) [reply .github/workflows/build.yml:109]: Considering the sqlite problem fix is merged, this is not required anymore

**Review by yqlbu — CHANGES_REQUESTED** [review]
Any test result?

File: .github/workflows/publish-docker-image.yml (hunk 4 above)
Comment (kunish) [reply .github/workflows/publish-docker-image.yml:25]: OK, noted, then github matrix is needed here

File: Dockerfile (hunk 5 above)
Comment (kunish) [reply Dockerfile:12]: I simply copy / pasted from dae

Comment (yqlbu) [reply Dockerfile:12]: Let's leave it as it is for now.

File: Dockerfile (hunk 6 above)
Comment (kunish) [reply Dockerfile:18]: OK, done

File: Dockerfile (hunk 5 above)
Comment (kunish) [reply Dockerfile:12]: OK

**Comment by yqlbu** [comment]
Also, consider adding a `.dockerignore` file to the project root.

**Review by yqlbu — APPROVED** [review]

**Comment by kunish** [comment]
Last success run log: https://github.com/daeuniverse/daed/actions/runs/5252280018/jobs/9488146854

**Review by mzz2017 — APPROVED** [review]

**Pull request merged**